website/docs/cloud-docs/api-docs/state-versions.mdx

```diff
@@ -48,12 +48,16 @@ Attribute                        | Description
 ---------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 `hosted-json-state-download-url` | A URL from which you can download the state data in a [stable format](/terraform/internals/json-format) appropriate for external integrations to consume. Only available if the state was created by Terraform 1.3+.
 `hosted-state-download-url`      | A URL from which you can download the raw state data, in the format used internally by Terraform.
+`hosted-json-state-upload-url`   | A URL to which you can upload the state data, in a [stable format](/terraform/internals/json-format) appropriate for external integrations to consume. Only available if the JSON state content has not been uploaded.
```

Comment:
regarding the last sentence, what would do the uploading? is this only an operator, or would TFC do this for you? and how do they generate the state data to upload?

Reply:
Yup this is a little tricky to word! We can clean this up a bit. Essentially there are two ways to upload: either by including the content in request body for `Create State Version` or, by omitting the content in that request and uploading the content to the URL provided in the response body `hosted-state-upload-url` and `hosted-json-state-upload-url`.